packages/@ionic/cli-utils/src/lib/project/angular/build.ts

```diff
@@ -1,15 +1,41 @@
+import { ParsedArgs, unparseArgs } from '@ionic/cli-framework';
 import chalk from 'chalk';
 import * as Debug from 'debug';
-
-import { ParsedArgs, unparseArgs } from '@ionic/cli-framework';
+import { CommandGroup, OptionGroup } from '../../../constants';
 
 import { AngularBuildOptions, CommandLineInputs, CommandLineOptions, CommandMetadata } from '../../../definitions';
-import { CommandGroup, OptionGroup } from '../../../constants';
 import { BUILD_SCRIPT, BuildRunner as BaseBuildRunner } from '../../build';
 import { addCordovaEngineForAngular, removeCordovaEngineForAngular } from './utils';
 
 const debug = Debug('ionic:cli-utils:lib:project:angular:build');
 
+// tslint:disable no-null-keyword
+export const NG_BUILD_OPTIONS = [
+  {
+    name: 'prod',
+    summary: `Flag to set configuration to ${chalk.green('prod')}`,
+    type: Boolean,
+    default: null,
```

Comment:
Reason for `default: null` ?

Reply:
I'm not sure if I grasped the way the CLI works here; this is what I was trying to achieve:

Say for example I had some additional metadata in `NG_BUILD_OPTIONS`
```
  {
    name: 'build-optimizer',
    summary: `Enable or disable the build optimizer`,
    type: Boolean,
    hint: 'ng',
  },
```
Then if I try to run
```
ionic build --prod
```
This is what the Angular CLI would receive
```
ng build --prod=true --build-optimizer=false
```
Which is bad, because `prod` is a meta flag that sets `build-optimizer=true`. I can't just set the default for `build-optimizer` to `true`, because running `ionic build` on it's own shouldn't enable the build optimizer.

So what I changed here, is line 76 in `@ionic/cli-framework/src/lib/options.ts`
```
    default: option.default !== undefined ? option.default : typeDefaults.get(type),
```
This stops the CLI using default values for options that are falsy. I then set the default value for the boolean options to `null` so I can control how they get passed through in `buildOptionsToNgArgs()`.

The end result is that the Ionic CLI won't pass through boolean flags to the Angular CLI if they have a default value of `null`.

So because of these changes and the way I changed `buildOptionsToNgArgs()` to filter args, it's necessary to set a default of `null` for the boolean options so that they don't default to `false`.

Reply:
Yeah, I think the issue is the filter/reduce that includes the flags with whatever value they have.

Boolean flags in the CLI default to `false` (meaning they were not specified on the command line). If they default to `true`, the CLI does some fancy work (mostly using behavior from minimist) to document the negation of the flag in help output. For example, with a boolean option `--foo` with `default: true`, the help output actually documents the negation: `--no-foo`. If `--no-foo` is specified on the command line, the `--foo` option is `false`, otherwise the value is taken from the default, true.

These changes would introduce something new: the differentiation between a boolean flag that is not set on the command line and one that defaults to `false`. It's probably worth doing because then command options would share a default of `null`, I'm just not certain what the implications of this change are without testing it. So, it's a little scary. The filter/reduce in is cool, but we lose type info and it's less explicit. I think you mostly had the filter/reduce because of the number of options, but now there aren't so many anymore.

Reply:
Yeah, I can take this out for now since those options are gone. It won't be a problem until it needs to differentiate between boolean options that are/aren't explicitly passed through the command line.